fdm-app/app/routes/farm.$b_id_farm.$calendar.balance.nitrogen._index.tsx

```diff
@@ -112,13 +116,17 @@ export default function FarmBalanceNitrogenOverviewBlock() {
     const location = useLocation()
     const _navigation = useNavigation()
     const page = location.pathname
-    const { nitrogenBalanceResult, farm, fields, errorMessage } = loaderData
+    const { farm, fields } = loaderData
+    const [asyncData, setAsyncData] = useState(loaderData.asyncData)
+    useEffect(() => setAsyncData(loaderData.asyncData), [loaderData.asyncData])
+
     const fieldsMap = new Map(fields.map((f) => [f.b_id, f]))
     return (
         <div className="space-y-4">
             <Suspense fallback={<NitrogenBalanceFallback />}>
-                <Await resolve={nitrogenBalanceResult}>
-                    {(resolvedNitrogenBalanceResult) => {
+                {asyncData.then(
```

Comment:
Why is the `.then` needed? I thought the React 19 documentation said that Suspense could handle resolving a promise itself.